libvast/src/system/index.cpp

```diff
@@ -1203,30 +1218,38 @@ index(index_actor::stateful_pointer<index_state> self,
       self->send(lookup.worker, atom::supervise_v, query_id, lookup.query,
                  std::move(actors),
                  caf::actor_cast<receiver_actor<atom::done>>(sink));
-      auto rp = self->make_response_promise<vast::uuid>();
+      auto rp = self->make_response_promise<partition_synopsis_pair>();
+      // TODO: Implement some kind of monadic composition instead of these
+      // nested requests.
```

Comment:
:smile:

Reply:
#2043 will at least make this a lot easier to read, so maybe we'll just switch to that once we fixed the remaining issues for GCC. :)